modules/standard/CTypes.chpl

```diff
@@ -298,8 +304,39 @@ module CTypes {
 
   // Note: we rely from nil to pointer types for ptr = nil, nil:ptr cases
 
+  /* Helper function for determining if casting between two types as pointee
+     types violates C strict aliasing rules. Only checks for types that have c_*
+     equivalents.
+   */
+  pragma "no doc"
+  inline proc pointeeCastStrictAliasingAllowed(type from, type to) param
+      : bool {
+    // allow identical types
+    if (from == to) {
+      return true;
+    }
+    // allow aliasing to any char type
+    if (to == int(8) || to == uint(8)) {
+      return true;
+    }
+    // allow types differing only in signedness
+    if (numBytes(from) == numBytes(to) &&
```

Comment:
`numBytes` is not available on all types, but it is available on the builtin numeric / primitive types. I'm recommending that you swap the order of the lhs and rhs of the `&&` to check that its real/integral before checking for the bytes.